### PR TITLE
Added the monitoring system to azure

### PR DESCRIPTION
**Pull request by athiruma**
## Type of change
Note: Fill **x** in []
- [ ] bug
- [x] enhancement
- [ ] documentation
- [x] dependencies

## Description

Phase 2:
Created the monitoring ticket attached resources.

### CODE REVIEW

File: cloud_governance/cloud_resource_orchestration/clouds/azure/resource_groups/monitor_cro_resources.py [hunk 1]
```diff
@@ -0,0 +1,54 @@
+from azure.mgmt.compute.v2023_03_01.models import VirtualMachine
+
+from cloud_governance.cloud_resource_orchestration.clouds.azure.resource_groups.abstract_resource import \
+    AbstractResource
+from cloud_governance.cloud_resource_orchestration.utils.constant_variables import DURATION, TICKET_ID
+
+
+class MonitorCROResources(AbstractResource):
+
```
Comment (ebattat) [inline cloud_governance/cloud_resource_orchestration/clouds/azure/resource_groups/monitor_cro_resources.py:9]: Missing documentation

File: cloud_governance/cloud_resource_orchestration/clouds/azure/resource_groups/monitor_cro_resources.py [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+    def __init__(self):
+        super().__init__()
+
+    def __monitor_instances(self):
+        """
+        This method the resources and return the data
```
Comment (ebattat) [inline cloud_governance/cloud_resource_orchestration/clouds/azure/resource_groups/monitor_cro_resources.py:15]: his method monitors resources and returns the data

File: cloud_governance/cloud_resource_orchestration/clouds/azure/resource_groups/monitor_cro_resources.py [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+                    'instance_name': virtual_machine.name,
+                    'email': self._compute_client.check_tag_name(tags=tags, tag_name='Email'),
+                    'duration': found_duration,
+                    'estimated_cost': self._compute_client.check_tag_name(tags=tags, tag_name='EstimatedCost')
+                })
+        return monitored_ticket_ids
+
+    def run(self):
+        """
+        This method start Azure CRO monitor
```
Comment (ebattat) [inline cloud_governance/cloud_resource_orchestration/clouds/azure/resource_groups/monitor_cro_resources.py:51]: starts

File: cloud_governance/cloud_resource_orchestration/clouds/azure/resource_groups/tag_cro_resources.py [hunk 4]
```diff
@@ -1,17 +1,20 @@
+from cloud_governance.cloud_resource_orchestration.clouds.azure.resource_groups.abstract_resource import \
+    AbstractResource
 from cloud_governance.cloud_resource_orchestration.utils.common_operations import get_ldap_user_data
 from cloud_governance.common.clouds.azure.compute.resource_group_operations import ResourceGroupOperations
 from cloud_governance.common.jira.jira_operations import JiraOperations
 from cloud_governance.common.logger.init_logger import logger
 
 
-class TagCROInstances:
+class TagCROResources(AbstractResource):
```
Comment (ebattat) [inline cloud_governance/cloud_resource_orchestration/clouds/azure/resource_groups/tag_cro_resources.py:10]: Missing documentation

File: cloud_governance/common/clouds/azure/compute/compute_operations.py [hunk 5]
```diff
@@ -11,12 +12,31 @@ def __init__(self):
...
         """
         This method returns all the compute resources by resource group
         :return:
         """
         instances_paged_object = self.__compute_client.virtual_machines.list(resource_group_name=resource_group_name)
-        instances_list = self._item_paged_iterator(item_paged_object=instances_paged_object)
+        instances_list: [VirtualMachine] = self._item_paged_iterator(item_paged_object=instances_paged_object)
         return instances_list
 
+    def get_all_instances(self) -> [VirtualMachine]:
```
Comment (ebattat) [inline cloud_governance/common/clouds/azure/compute/compute_operations.py:24]: Missing documentation

**Review by ebattat — COMMENTED** [review]
Do we need to add any test ?

**Comment by athiruma** [comment]
> Do we need to add any test ?

Yeah, will add the mock unit tests.

**Review by ebattat — COMMENTED** [review]
Do we need to add any unittest ?

**Comment by athiruma** [comment]
> Do we need to add any unittest ?

Yeah, will add the mock unit tests. I will add the unit tests later.

**Review by ebattat — APPROVED** [review]

**Pull request merged**